server/src/main/java/org/elasticsearch/index/analysis/PreBuiltAnalyzerProviderFactory.java

```diff
@@ -48,15 +48,14 @@ public class PreBuiltAnalyzerProviderFactory extends PreConfiguredAnalysisCompon
         super(name, new PreBuiltAnalyzersDelegateCache(name, preBuiltAnalyzer));
         this.create = preBuiltAnalyzer::getAnalyzer;
         Analyzer analyzer = preBuiltAnalyzer.getAnalyzer(Version.CURRENT);
-        analyzer.setVersion(Version.CURRENT.luceneVersion);
+        // there are no calls to getVersion(), maybe these can be removed
```

Comment:
See https://github.com/apache/lucene/commit/1e7d8146fff95522674c720099a52b9efd102881

Looks like this would need to be set as part of a CustomAnalyzer.builder, but I'm not sure if we even utilized this functionality fully. 

Maybe @seut knows more.

Reply:
It says the version defaults to `LATEST`.
https://github.com/apache/lucene/blob/5b92002fed3ca316e98c822c1afdccd30f00feb7/lucene/analysis/common/src/java/org/apache/lucene/analysis/custom/CustomAnalyzer.java#L243-L244

And from the commit, there is a test building `CustomAnalyzer` without setting the version. I think this should be fine.